src/Orleans.Core.Abstractions/Manifest/GrainInterfaceId.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Diagnostics;
+
+namespace Orleans.Metadata
+{
+    /// <summary>
+    /// Uniquely identifies a grain interface.
+    /// </summary>
+    [Serializable]
+    [DebuggerDisplay("{Value}")]
+    public readonly struct GrainInterfaceId : IEquatable<GrainInterfaceId>
```

Comment:
IMO naming is too similar to `GrainId`. What about `GrainInterfaceTypeCode` or something like that?

Reply:
I'm not a fan of the TypeCode name, especially the "code" part, but I'm open to suggestions. It could be `GrainInterfaceDescriptor` if you prefer.